### PR TITLE
feat: Filter, search and Home page 

### PR DESCRIPTION
**Pull request by Chakravarthi0**
Implemented filter and search:
- user can filter videos by category and upload date. 
- user can search a video by video title or channel name.

Refactored homepage.

### CODE REVIEW

**Review by manojkumar8400 — APPROVED** [review]
UI looks good and all features are working properly.

File: src/components/search-bar/search-bar.css [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+.explore-nav > input {
+  background-color: var(--white-pure);
+}
+
+@media screen and (max-width: 950px) {
+  .nav-search {
+    display: none;
+  }
+  .explore-nav {
+    display: flex !important;
```
Comment (Kothalkarpriya) [inline src/components/search-bar/search-bar.css]: Less use of !important might be better

File: src/pages/explore/explore.css [hunk 2]
```diff
@@ -1,3 +1,28 @@
...
+}
+
+.filter-container{
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+    margin: 0 2rem;
+}
+
+
```
Comment (Kothalkarpriya) [inline src/pages/explore/explore.css:12]: remove extra space

**Review by Kothalkarpriya — APPROVED** [review]
Look into my suggestions once, otherwise, your code is clean and understandable. You can merge changes

File: src/components/search-bar/search-bar.css (hunk 1 above)
Comment (Chakravarthi0) [reply src/components/search-bar/search-bar.css]: I forgot to remove "!important", it is unnecessary. Thanks for pointing it out. removed.

**Pull request merged**